Lib/_strptime.py

```diff
@@ -577,3 +582,27 @@ def _strptime_datetime(cls, data_string, format="%a %b %d %H:%M:%S %Y"):
         args += (tz,)
 
     return cls(*args)
+
+def _strptime_datetime_date(data_string, format):
+    """Return a date based on the input string and the format string."""
+    msg = "'{!s}' {} not valid in date format specification."
+    if _check_invalid_datetime_specs(format, time_specs, msg):
+        _date = _strptime_datetime(datetime_datetime, data_string, format)
+        return _date.date()
+
+def _strptime_datetime_time(data_string, format):
+    """Return a time based on the input string and the format string."""
+    msg = "'{!s}' {} not valid in time format specification."
+    if _check_invalid_datetime_specs(format, date_specs, msg):
+        _time = _strptime_datetime(datetime_datetime, data_string, format)
+        return _time.time()
+
+def _check_invalid_datetime_specs(fmt, blacklist_specs, msg):
+    found_invalid_specs = []
+    for spec in blacklist_specs:
+        if spec in fmt:
+            found_invalid_specs.append(spec)
+    if found_invalid_specs:
+        suffix = "are" if len(found_invalid_specs) > 1 else "is"
```

Comment:
nitpick: English is not my first language, so I'm not sure, but the plural rule should be len() != 1, no? ... it's just a theorical question, in practical len cannot be zero :-D Sorry.

Reply:
@vstinner As a native speaker, I think you are right:

"There are zero apples"
"There is one apple"
"There are two apples"

Though in this case we can just get rid of the "suffix" concept entirely, because the message is:

```
msg = "'{!s}' {} not valid in date format specification.`
```

It's fine to just say "x, y and z not found in date format specification" in an error message.